sdk/src/main/java/io/customer/sdk/di/CustomerIOComponent.kt

```diff
@@ -39,47 +39,41 @@ import java.util.concurrent.TimeUnit
 /**
  * Configuration class to configure/initialize low-level operations and objects.
  */
-internal class CustomerIOComponent(
-    private val customerIOConfig: CustomerIOConfig,
-    private val context: Context
-) {
+class CustomerIOComponent : DiGraph() {
 
-    val sdkConfig: CustomerIOConfig
-        get() = customerIOConfig
-
-    val siteId: String
-        get() = customerIOConfig.siteId
+    lateinit var context: Context
```

Comment:
I am not a bit divided on this, on removing these from constructors to `lateinit`. I feel like this would make code a bit error-prone, in case someone doesn't set these values right after initializing `CustomerIOComponent`

I am bit inclined on having values as part of constructors, that just makes it modular, enforces user to provide values to must-have parameters.

Reply:
Same as [this comment](https://github.com/customerio/customerio-android/pull/68#discussion_r829204969), this change was made to make an empty constructor of `CustomerIOComponent`. Since this is no longer a requirement, I will change it back. I agree `lateinit` is more error-prone and it's a good idea to avoid it when there is a good alternative.